src/backend/kvm/thread.rs

```diff
@@ -77,35 +75,21 @@ impl<P: KeepPersonality> Thread<P> {
             .as_virt()
             .start;
 
-        Ok([vaddr.as_u64().into(), 0.into()])
+        Ok(vaddr.as_u64() as _)
     }
 
-    pub fn meminfo(&self, block: &mut Block) -> Result<[Register<usize>; 2], i32> {
+    pub fn meminfo(&self) -> Result<usize, i32> {
         let keep = self.keep.read().unwrap();
 
         // The maximum number of memory slots possible for a virtual machine
         // minus the ones which were already used.
         let mem_slots = keep.kvm_fd.get_nr_memslots() - keep.regions.len();
-
-        // FIXME:
-        // Obsolete, if [host side syscall verification and address translation](https://github.com/enarx/enarx/issues/957)
-        // is implemented.
-        let virt_start = Address::from(keep.sallyport_start.as_mut_ptr());
-
-        let mem_info: MemInfo = MemInfo {
-            virt_start,
-            mem_slots,
-        };
-
-        let c = block.cursor();
-        c.write(&mem_info).map_err(|_| libc::ENOBUFS)?;
-
-        Ok([0.into(), 0.into()])
+        Ok(mem_slots)
```

Comment:
nit; should we just avoid the intermediate variable declaration here?

Reply:
done